### PR TITLE
Fix to rh refactoring as parking was breaking beam

### PR DESCRIPTION
**Pull request by haitamlaarabi**
Another fix

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3256)

### CODE REVIEW

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Review by nikolayilyin — APPROVED** [review]
don't see anything suspicious

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test Now!

**Comment by haitamlaarabi** [comment]
Test!

**Comment by haitamlaarabi** [comment]
Test now!

File: src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala [hunk 1]
```diff
@@ -725,117 +707,108 @@ class RideHailAgent(
...
         beamLegsToIgnoreDueToNewPassengerSchedule.add(beamLeg)
       }
       stash()
       goto(OfflineInterrupted)
     case ev @ Event(UnhandledVehicle(_, _, _), _) =>
-      log.debug(s"state(RideHailingAgent.WaitingToDriveInterrupted.UnhandledVehicle): $ev")
+      log.debug(s"state(RideHailingAgent.WaitingToDriveInterrupted.UnhandledVehicle): $ev, Vehicle ID: ${vehicle.id}")
       stash()
-      stay()
+      goto(IdleInterrupted)
```
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala:768]: Why is this now going to IdleInterrupted?

File: src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala [hunk 2]
```diff
@@ -442,10 +441,10 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash with Expon
       log.debug("state(DrivesVehicle.Driving.StartingRefuelSession): {}", ev)
       stay()
     case ev @ Event(UnhandledVehicle(_, _, _), _) =>
-      log.debug("state(DrivesVehicle.Driving.UnhandledVehicle): {}", ev)
+      log.error("state(DrivesVehicle.Driving.UnhandledVehicle): {}", ev)
```
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala:444]: Should these still be error, or were they for debugging more easily?

**Review by JustinPihony — APPROVED** [review]
Nothing major to block - definitely curious about the parallelism though

**Pull request merged**